Added `attention_jax.ipynb` for `attention_torch.ipynb`

patel-zeel opened the pull request:
Based on: [convert pytorch demos to JAX](https://github.com/probml/pyprobml/issues/686).

* Converted `torch` code to `jax` and `flax`.
* Verified that it runs on colab (after manually installing `flax`).

Edit:
* [Colab link](https://colab.research.google.com/github/patel-zeel/probml-notebooks/blob/attention_jax/notebooks-d2l/attention_jax.ipynb)
* [Rendered images](https://gist.github.com/patel-zeel/28a101009f5e938082b6d93101bf1fd7)
* [ReviewNB link](https://app.reviewnb.com/probml/probml-notebooks/pull/40/)

murphyk commented:
- Please add a link to a version of the file which I can run
- Please include commands to install flax etc
- Please include any rendered images in the output

patel-zeel commented:
> * Please add a link to a version of the file which I can run

I have added the [colab link](https://colab.research.google.com/github/patel-zeel/probml-notebooks/blob/attention_jax/notebooks-d2l/attention_jax.ipynb).
> * Please include commands to install flax etc

Added.
> * Please include any rendered images in the output

Added.
